src/toil/common.py

```diff
@@ -255,6 +260,18 @@ def _addOptions(addGroupFn, config):
                      'that do not specify an explicit value for this requirement. Standard '
                      'suffixes like K, Ki, M, Mi, G or Gi are supported. Default is %s' %
                      bytes2human( config.defaultDisk, symbols='iec' ))
+    addOptionFn("--useSharedCache", dest="useSharedCache", action='store_true', default=None,
+                help=('Should the shared cache strategy be used? If true, all workers on a node '
+                      'will share a cache and parallel workers requesting the same file will link '
+                      'to the same cached file thereby reducing the disk used, and increasing '
+                      'speed of file operations. This overrides the defaultCache parameter. '
+                      'Default is False'))
+    addOptionFn("--backwardsCompatible", dest="backwardsCompatible", action='store_true',
+                default=None, help='Shared caching disallows modification of read global files by '
+                'default and thus is backwards incompatible with scripts written in Toil versions '
+                'lower than 3.2.0. This flag makes it compatible with those scripts, however it '
+                'also defeats the purpose of shared caching via hard links to save space. Default '
+                'is False')
```

Comment:
Definitely think we should get rid of this in a pull request straight after this one - if not now.